eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -65,7 +68,8 @@ case class SignedMessage(nodeId: PublicKey, message: String, signature: ByteVect
 
 case class VerifiedMessage(valid: Boolean, publicKey: PublicKey)
 
-case class SendOnionMessageResponse(sent: Boolean, failureMessage: Option[String])
+case class ResponsePayload(replyPath: Option[String], unknownTlvs: Map[String, ByteVector])
+case class SendOnionMessageResponse(failureMessage: Option[String], response: Option[ResponsePayload])
```

Comment:
This new data contract makes it harder to use for fire-and-forget onion messages (which could be used for messaging for example), as they would get a completely empty response when the message is correctly sent. I think we should keep the `sent` boolean for those use-cases, don't you think?

Similarly to my comment on the `OnionMessageReceived`, I think it's useful to keep the reply path in json to allow easy inspection, we can have both the json and wire-encoded versions in the response, can't we?

Apart from that, `ResponsePayload` is too broad for a case class that's put on the top-level namespace, what about:
```suggestion
case class SendOnionMessageResponsePayload(encodedReplyPath: Option[String], replyPath: Option[BlindedRoute], unknownTlvs: Map[String, ByteVector])
case class SendOnionMessageResponse(sent: Boolean, response: Option[SendOnionMessageResponsePayload], failureMessage: Option[String])
```